mlrun/api/db/sqldb/db.py

```diff
@@ -651,51 +639,92 @@ def list_tags(self, session, project: str):
                 tags.add(tag.name)
         return tags
 
-    def add_project(self, session, project: dict):
-        project = project.copy()
-        name = project.get("name")
-        if not name:
-            raise ValueError("project missing name")
-
-        project.pop("users", [])
-        prj = Project(**project)
-        users = (
-            []
-        )  # self._find_or_create_users(session, user_names) - user_names are previously popped users
-        prj.users.extend(users)
-        self._upsert(session, prj)
-        self._projects.add(prj.name)
-        return prj.id
-
-    def update_project(self, session, name, data: dict):
-        prj = self.get_project(session, name)
-        if not prj:
-            raise DBError(f"unknown project - {name}")
-
-        data = data.copy()
-        data.pop("users", [])
-        for key, value in data.items():
-            if not hasattr(prj, key):
-                raise DBError(f"unknown project attribute - {key}")
-            setattr(prj, key, value)
-
-        users = (
-            []
-        )  # self._find_or_create_users(session, user_names) - user_names are previously popped users
-        prj.users.clear()
-        prj.users.extend(users)
-        self._upsert(session, prj, ignore=True)
-
-    def get_project(self, session, name=None, project_id=None):
-        if not (project_id or name):
-            raise ValueError("need at least one of project_id or name")
-
-        if project_id:
-            return self._query(session, Project).get(project_id)
-
-        return self._query(session, Project, name=name).one_or_none()
-
-    def delete_project(self, session, name: str):
+    def create_project(self, session: Session, project: schemas.Project):
+        self._create_project(session, project)
+
+    def _create_project(self, session: Session, project: schemas.Project):
```

Comment:
I removed it, but as a general rule:
I prefer to keep any formatting rules out unless they are done & verified automatically, i.e. unless we have a formatter that is doing this method order automatically (and verifying it in CI) it doesn't worth the hassle IMO

Reply:
Well i hate to ruin the "good life" you have, but you asked for a review :)
Keeping public methods first and private after is something I believe in and has proved itself over the years over many projects. It's quite unanimous (don't recall ever hearing an argument against it from anyone personally). Saying that because it is not being done automatically for you then it's not a worthy rule to stick by is problematic. I kindly suggest you consider this, as it greatly helps readability and orientation in large bodies of code.
Do with this information you wish